code/controllers/subsystems/ticker.dm

```diff
@@ -322,6 +330,47 @@ var/datum/controller/subsystem/ticker/SSticker
 
 	return 1
 
+/datum/controller/subsystem/ticker/proc/setup_player_ready_list()
+	// One big loop at the start once SSticker comes up to catch anyone who readied before-hand
+	// Afterward the list updates on a per-user basis in menu.dm and logout.dm
+	if(current_state >= GAME_STATE_PLAYING || !SSjobs)
+		return // how
+
+	for(var/mob/abstract/new_player/NP in player_list)
+		if(NP.ready)
+			update_ready_list(NP)
+
+	total_players_ready = LAZYLEN(ready_player_jobs)
+
+/datum/controller/subsystem/ticker/proc/update_ready_list(var/mob/abstract/new_player/NP)
+	if(current_state >= GAME_STATE_PLAYING || !SSjobs || !istype(NP))
```

Comment:
```suggestion
	if(current_state >= GAME_STATE_PLAYING || !SSjobs || !istype(NP))
```

Not a huge thing buuuut:

We're already checking `istype(NP)` before this proc is called every time. Don't need to do it within this proc as well as there.